plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -1,19 +1,182 @@
 package com.google.cloud.tools.eclipse.appengine.deploy.ui;
 
+
 import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Control;
 import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
 import org.eclipse.ui.IWorkbenchPropertyPage;
 import org.eclipse.ui.dialogs.PropertyPage;
 
 public class DeployPropertyPage extends PropertyPage implements IWorkbenchPropertyPage {
 
+  private Text version;
+  private Text projectId;
+  private Text bucket;
+  private Button defaultVersionButton;
+  private Button customVersionButton;
+  private Button autoPromoteButton;
+  private Button manualPromoteButton;
+  private Button defaultBucketButton;
+  private Button customBucketButton;
+
+  private RadioSelectionListener versionSelectionListener;
+  private RadioSelectionListener bucketSelectionListener;
+
+  public class RadioSelectionListener implements SelectionListener {
+
+    private Text text;
+    private Button button;
+
+    public RadioSelectionListener(Text text, Button button) {
+      this.text = text;
+      this.button = button;
+    }
+
+    @Override
+    public void widgetSelected(SelectionEvent e) {
+      doSelection(e);
+    }
+
+    @Override
+    public void widgetDefaultSelected(SelectionEvent e) {
+      doSelection(e);
+    }
+
+    private void doSelection(SelectionEvent e) {
+      text.setEnabled(e.getSource() == button);
+    }
+  }
+
   @Override
   protected Control createContents(Composite parent) {
-    Label label = new Label(parent, SWT.LEFT);
-    label.setText(Messages.getString("preferences.title")); //$NON-NLS-1$
-    return label;
+    Composite container = new Composite(parent, SWT.NONE);
```

Comment:
This method is very long. Is it feasible to split into smaller helper methods, purely for clarity?